### PR TITLE
Feature supply share constraint

### PR DESCRIPTION
**Pull request by timtroendle**
Reviewer checklist:

- [x] Test(s) added to cover contribution
- [x] Documentation updated
- [x] Changelog updated
- [x] Coverage maintained or improved

### CODE REVIEW

File: calliope/backend/pyomo/constraints/group.py [hunk 1]
```diff
@@ -79,3 +91,52 @@ def demand_share_constraint_rule(backend_model, constraint_group, carrier, what)
...
+def supply_share_constraint_rule(backend_model, constraint_group, carrier, what):
+    """
+    Enforces shares of carrier_prod for groups of technologies and locations. The
+    share is relative to ``supply`` and ``supply_plus`` technologies only.
+
+    .. container:: scrolling-wrapper
+
+        .. math::
+
+            \\sum_{loc::tech::carrier \\in given\\_group, timestep \\in timesteps} carrier_{prod}(loc::tech::carrier, timestep) <= / >=
```
Comment (timtroendle) [inline calliope/backend/pyomo/constraints/group.py]: Can I build that locally? I have no idea of the syntax and whether this is correct.

Comment (brynpickering) [reply calliope/backend/pyomo/constraints/group.py]: If you build the docs (`build html` when in `calliope/docs`) then it should be possible to see the result of that in the locally built html files. Needs a bunch of dependencies installed to build the docs (see the relevant requirements file)

Comment (timtroendle) [reply calliope/backend/pyomo/constraints/group.py]: cheers, I'll give it a try

**Comment by brynpickering** [comment]
Note that constraints are no longer built over `backend_model.constraint_groups`, but rather `backend_model.group_names_NAME_OF_CONSTRAINT`. This ensures that Pyomo doesn't try to build a constraint over a group that it doesn't fit within.

**Comment by brynpickering** [comment]
Note that constraints are no longer built over `backend_model.constraint_groups`, but rather `backend_model.group_names_NAME_OF_CONSTRAINT`. This ensures that Pyomo doesn't try to build a constraint over a group that it doesn't fit within.

**Comment by timtroendle** [comment]
Does that impact line 32 and 38 in groups.py?

I was actually surprised to see that I did not have to touch the implementation for the tests to pass. I assume that because I can append constraints to the backend where ever I want, but when adding them to `backend_model.group_names_NAME_OF_CONSTRAINT` there is a special mechanism behind?

**Comment by brynpickering** [comment]
Yep, the new implementation doesn't actually invalidate the use of the old method. However, I would expect the old method to still fail in circumstances where there are more complex combinations of group constraints (which is why the implementation was updated).

**Comment by sjpfenninger** [comment]
@timtroendle Once you fix the failing test and resolve conflicts I will merge this

**Pull request merged**